src/workers/filesystems/infrastructure/SyncLocalFileSystem.ts

```diff
@@ -0,0 +1,341 @@
+                emptyFileError,
+                'Reading metadata of file',
+                `Size of file is ${metadata.size}`
+              ),
+            });
+          }
+        } catch (e) {
+          const err = e as { code?: string };
+
+          readingMetaErrors.push({
+            name: relativeName,
+            errorName: err.code === 'EPERM' ? 'NO_PERMISSION' : 'UNKNOWN',
+            errorDetails: createErrorDetails(
+              err,
+              'Reading metadata of file',
+              `File name is ${itemName}`
+            ),
+          });
+        }
+      }
+      return { listing, readingMetaErrors };
+    },
+
+    async deleteFile(name: string) {
+      const actualPath = getActualPath(name);
+      try {
+        await fs.unlink(actualPath);
+      } catch (e) {
+        const err = e as { code?: string };
+
+        if (err.code !== 'ENOENT')
+          throw new ProcessError(
+            'UNKNOWN',
+            createErrorDetails(err, 'Deleting a file locally', `Name: ${name}`)
+          );
+      }
+    },
+
+    pullFile(name: string, source: Source) {
+      return new Promise(async (resolve, reject) => {
+        const tmpFilePath = getTempFilePath();
+
+        const { stream: sourceFile, ...sourceWithoutStream } = source;
+
+        Logger.debug(`Downloading ${name} to temp location ${tmpFilePath}`);
+
+        const destination = createWriteStream(tmpFilePath);
+
+        try {
+          await pipeline(sourceFile, destination);
+
+          const actualPath = getActualPath(name);
+
+          await fs.mkdir(path.parse(actualPath).dir, {
+            recursive: true,
+          });
+
+          await saferRenameFile(tmpFilePath, actualPath);
+
+          const modTime = getDateFromSeconds(source.modTime);
+          fs.utimes(actualPath, modTime, modTime);
+
+          resolve();
+        } catch (err: unknown) {
+          reject(
+            new ProcessError(
+              'UNKNOWN',
+              createErrorDetails(
+                err,
+                `Pulling file locally`,
+                `Name: ${name}, source: ${JSON.stringify(
+                  sourceWithoutStream,
+                  null,
+                  2
+                )}`
+              )
+            )
+          );
+        }
+      });
+    },
+
+    async pullFolder(name): Promise<void> {
+      const osSpecificRelative = name.replaceAll('/', path.sep);
+      const fullPath = path.join(localPath, osSpecificRelative);
+
+      await fs.mkdir(path.parse(fullPath).dir, { recursive: true });
+    },
+
+    renameFile(oldName: string, newName: string) {
+      const oldActualPath = getActualPath(oldName);
+      const newActualPath = getActualPath(newName);
+      return saferRenameFile(oldActualPath, newActualPath);
+    },
+
+    async existsFolder(name: string): Promise<boolean> {
+      const actualPath = getActualPath(name);
+      try {
+        await fs.access(actualPath);
```

Comment:
why not use `fs. existsSync ` ?

Reply:
Not sure why `fs.access` was chosen. The only thing it comes to me is to ignore system folders